nimpy.nim

```diff
@@ -767,7 +770,13 @@ macro callNimProcWithPythonArgs(prc: typed, argsTuple: PPyObject): PPyObject =
         updateStackBottom()
         if verifyArgs(`argsTuple`, `argsLen`, `argsLenReq`, `nameLit`):
             `parseArgsStmts`
-            nimValueToPy(`origCall`)
+            try:
+                nimValueToPy(`origCall`)
+            except:
+                let e = getCurrentException()
+                let err = pyLib.PyErr_NewException("nimpy" & "." & $(e.name), pyLib.NimPyException, nil)
+                pyLib.PyErr_SetString(err, "Unexpected error encountered: " & getCurrentExceptionMsg())
```

Comment:
Is `err` leaking?

Reply:
I had a similar question, I believe so. `PyErr_NewException` creates a new Exception Python class object and increments the reference by 1, so I think it does result in a leak technically. Any suggestions on what do in this situation? Should we store the exception somewhere?

Reply:
I think `decRef err` should work here.

Reply:
I made the change and tests pass locally.